stackstac/rio_reader.py

```diff
@@ -343,18 +342,30 @@ def _open(self) -> ThreadsafeRioDataset:
                     "a separate STAC asset), so you'll need to exclude this asset from your analysis."
                 )
 
+            ds_epsg: int | None
+            try:
+                ds_epsg = ds.crs.to_epsg() if ds.crs is not None else None
+            except rasterio.errors.CRSError:
+                ds_epsg = None
+
             # Only make a VRT if the dataset doesn't match the spatial spec we want
-            if self.spec.vrt_params != {
-                "crs": ds.crs.to_epsg(),
+            if ds_epsg is None or self.spec.vrt_params != {
+                "crs": ds_epsg,
                 "transform": ds.transform,
```

Comment:
MIght need to set [`src_transform`](https://rasterio.readthedocs.io/en/latest/api/rasterio.vrt.html#rasterio.vrt.WarpedVRT) in addition to `transform` as mentioned by @vincentsarago in https://github.com/gjoseph92/stackstac/issues/181#issuecomment-1373918159?

Reply:
So looking at the code it might be a bit more complex. 

What I do in rio-tiler, is automatically create a WarpedVRT when we have gcps (which in stackstac case should replace the `ds`) and then I create another WarpedVRT on top of it if I need warping